paymentsheet/src/main/java/com/stripe/android/link/repositories/LinkApiRepository.kt

```diff
@@ -249,24 +250,29 @@ internal class LinkApiRepository @Inject constructor(
             mapOf("billing_phone" to it)
         } ?: emptyMap()
 
+        val paymentMethodParams = mapOf("expand" to listOf("payment_method"))
+
         stripeRepository.sharePaymentDetails(
             consumerSessionClientSecret = consumerSessionClientSecret,
             id = id,
             extraParams = mapOf(
                 "payment_method_options" to extraConfirmationParams(paymentMethodCreateParams.toParamMap()),
-            ) + allowRedisplay + billingPhone,
+            ) + allowRedisplay + billingPhone + paymentMethodParams,
             requestOptions = buildRequestOptions(),
         ).onFailure {
             errorReporter.report(ErrorReporter.ExpectedErrorEvent.LINK_SHARE_CARD_FAILURE, StripeException.create(it))
-        }.map { passthroughModePaymentMethodId ->
+        }.map { paymentMethod ->
+            val paymentMethodId = requireNotNull(paymentMethod.id)
             LinkPaymentDetails.Saved(
                 paymentDetails = ConsumerPaymentDetails.Passthrough(
```

Comment:
This `ConsumerPaymentDetails.Passthrough` needs to go. I have a [draft pull request](https://github.com/stripe/stripe-android/pull/11064) up to get rid of it, but haven't found the time to work on it again. Will rebase on this branch so that we can hopefully get rid of this soon!